### DISCUSSION TITLE
What does the garbage collector do View objects?

### DISCUSSION THREAD

**Discussion by CanavarB** [Rapptz/discord.py#8976] [Q&A]
I designed a bot about ID check. I need to ask users to their IDs with bot from DM Channel. I'm using buttons but every view object is creating again when I'm sending a new message with view object. I need to clear previous one of them from RAM.

**Comment by Rapptz** [comment]
Views are automatically cleaned up when they're stopped or timeout is reached.

**Reply by CanavarB** [reply]
I thought so, too. That's why I always kept their timeout value low. But I realized these view object s "__del__" method is not called when they stoped or timed out. When I terminated to whole process, these methods are executing.
